pkg/controller/service/service_controller.go

```diff
@@ -582,7 +591,45 @@ func nodeSlicesEqualForLB(x, y []*v1.Node) bool {
 	return nodeNames(x).Equal(nodeNames(x))
 }
 
-func getNodeConditionPredicate() corelisters.NodeConditionPredicate {
+func getNodeTaintPredicate() corelisters.NodePredicate {
+	return func(node *v1.Node) bool {
+		// We add the master to the node list, but its unschedulable.  So we use this to filter
+		// the master.
+		if node.Spec.Unschedulable {
+			return false
+		}
+		// If no taint, accept
+		if len(node.Spec.Taints) == 0 {
+			return true
+		}
+		// As of 1.6, we will taint the master, but not necessarily mark it unschedulable.
```

Comment:
Did it ever happen?

Reply:
I still check if a node is a master using label to exclude it from the pool, kubadm is setting a taint NoSchedule to the master so master will be exclude during the check for effect NoSchedule or NoExecute

Reply:
We can't have a NoSchedule nor NoExecute filter here. Pods can tolerate Taints, and they still need to be able to handle traffic.

Reply:
I understand, so is there a way actually to know if a pod which tolerate a taint is schedule on the node ?

Reply:
For NoSchedule it doesn't even need to tolerate this Taint, as NoSchedule applies only to Pods that will want to schedule on the given one after the Taint is added.